### PR TITLE
Fix search around

### LINKED ISSUE #364: Query search_around_coordinates broken
```
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
/tmp/ipykernel_90/1668774235.py in <module>
      5     ten_arcsec_query.find_fields()
      6     ten_arcsec_query.fields_df.field = ten_arcsec_query.fields_df.field.str.replace('RACS','VAST')
----> 7     ten_arcsec_query.find_sources()
      8     display(ten_arcsec_query.results)

/opt/conda/lib/python3.9/site-packages/vasttools/query.py in find_sources(self)
   1177 
   1178         if self.settings['search_around']:
-> 1179             meta['index'] = 'i'
   1180 
   1181         self.logger.debug("Getting components...")

UnboundLocalError: local variable 'meta' referenced before assignment
```

### PR DESCRIPTION
**Pull request by ddobie**

### CODE REVIEW

File: vasttools/query.py [hunk 1]
```diff
@@ -1388,11 +1388,7 @@ def _init_sources(self, group: pd.DataFrame) -> Source:
             source_image_type = "COMBINED"
         source_islands = self.settings['islands']
 
-        source_df = group.drop(
-            columns=[
-                '#'
-            ]
-        )
+        source_df = group.drop(group.filter('#'))
```
Comment (ajstewart) [inline vasttools/query.py:1391]: What's happening here with the filter within the drop? I can't remember what the dataframe exactly looks like, and I'm unsure what that filter is returning.

Comment (ddobie) [reply vasttools/query.py:1391]: It's just dropping the '#' column if it exists. `df.drop` throws an error if you try and drop a column that doesn't exist and I think this is slightly nicer than having an if statement. However, now that I think about it maybe it's less efficient computationally?

Comment (ajstewart) [reply vasttools/query.py:1391]: There's probably minimal difference performance wise, I was just unsure as to what is actually going on here, if you do:
```
In [11]: a
Out[11]:
   a  b  #
0  1  1  1
1  2  2  2
2  3  3  3

In [12]: b = a.drop(a.filter('#'))
```

Then it errors for me:

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-12-6dd58ae211d1> in <module>
----> 1 b = a.drop(a.filter('#'))

~/anaconda3/envs/vast-tools-dev-v2/lib/python3.8/site-packages/pandas/util/_decorators.py in wrapper(*args, **kwargs)
    309                     stacklevel=stacklevel,
    310                 )
--> 311             return func(*args, **kwargs)
    312
    313         return wrapper

~/anaconda3/envs/vast-tools-dev-v2/lib/python3.8/site-packages/pandas/core/frame.py in drop(self, labels, axis, index, columns, level, inplace, errors)
   4899                 weight  1.0     0.8
   4900         """
-> 4901         return super().drop(
   4902             labels=labels,
   4903             axis=axis,

~/anaconda3/envs/vast-tools-dev-v2/lib/python3.8/site-packages/pandas/core/generic.py in drop(self, labels, axis, index, columns, level, inplace, errors)
   4145         for axis, labels in axes.items():
   4146             if labels is not None:
-> 4147                 obj = obj._drop_axis(labels, axis, level=level, errors=errors)
   4148
   4149         if inplace:

~/anaconda3/envs/vast-tools-dev-v2/lib/python3.8/site-packages/pandas/core/generic.py in _drop_axis(self, labels, axis, level, errors)
   4180                 new_axis = axis.drop(labels, level=level, errors=errors)
   4181             else:
-> 4182                 new_axis = axis.drop(labels, errors=errors)
   4183             result = self.reindex(**{axis_name: new_axis})
   4184

~/anaconda3/envs/vast-tools-dev-v2/lib/python3.8/site-packages/pandas/core/indexes/base.py in drop(self, labels, errors)
   6016         if mask.any():
   6017             if errors != "ignore":
-> 6018                 raise KeyError(f"{labels[mask]} not found in axis")
   6019             indexer = indexer[~mask]
   6020         return self.delete(indexer)

KeyError: "['#'] not found in axis"
```

Which is what I expected as both these are acting on the index axis and not the column axis.

```
In [13]: b = a.drop(a.filter('#'), axis=1)

In [14]: b
Out[14]:
   a  b
0  1  1
1  2  2
2  3  3
```

You could do:
```
In [15]: c = b.drop('#', axis=1, errors='ignore')
```

But I'm not confident enough to have it ignore all errors. From a simplicity standpoint I'd probably just go for a:

```
if `#' in x.columns:
    # drop it
```

But I'm scratching my head as to how it isn't erroring when this line is in?

Comment (ddobie) [reply vasttools/query.py:1391]: How odd...

It works for me and as far as I can tell it's covered by the testing, so it should definitely be throwing an error if there was one. I agree it's probably better to switch to an if statement though, and I've done so.

**Review by ajstewart — APPROVED** [review]

**Pull request merged**